application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -252,8 +251,11 @@ private RestAction<Message> dmUser(MessageChannelUnion sourceChannel,
     }
 
     private RestAction<Void> deleteOriginalMessage(JDA jda, String channelId, String messageId) {
-        TextChannel sourceChannel = Objects.requireNonNull(jda.getTextChannelById(channelId),
-                "Source channel could not be found for transfer-question feature");
+        MessageChannel sourceChannel = jda.getTextChannelById(channelId);
+
+        if (sourceChannel == null) {
+            sourceChannel = jda.getThreadChannelById(channelId);
+        }
```

Comment:
we'll have to pass a channel type in [getChannelById(Class<T> type, long id)](https://javadoc.io/doc/net.dv8tion/JDA/latest/net/dv8tion/jda/api/JDA.html#getChannelById(java.lang.Class,long)) as well, there is no general method to just get the channel, even if there was the `Channel` interface have no delete message methods because, it is used to represent audio channels as well. On be better side, thread and text are the only type of channel anyone will transfer a question  from, so we're  fine for now

Reply:
`jda.getChannelById(MessageChannel.class, channelId)`

there, this supports both, text and thread channels, and has `delete...`.

Reply:
done